### ISSUE TITLE
style at Head not work in Shape

### ISSUE THREAD

**Issue by devenliu** [OrchardCMS/OrchardCore#13887]
### Describe the bug
<style at="Head"> block defined in Shape is not rendered 

### To Reproduce
1. Create a 'MyShape.cshtml'
2. Define <style at="Head">
3. Run

MyShape.cshtml
```
<div class="myshape">MyShape Html</div>

<style at="Head">
    .myshape {
        color: red;
        font-size: 18px;
    }
</style>
```

Layout.cshtml
```
<!DOCTYPE html>
<html lang="@Orchard.CultureName()">
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>@RenderTitleSegments(Site.SiteName, "before")</title>
    <link type="image/x-icon" rel="shortcut icon" href="~/TheTheme/favicon.ico">
    <script asp-name="bootstrap" version="5" at="Foot"></script>
    <style asp-name="thetheme-bootstrap-oc" version="1"></style>
    <style asp-name="font-awesome" version="6"></style>
    <resources type="Meta" />
    <resources type="HeadLink" />
    <resources type="Stylesheet" />
    <resources type="HeadScript" />
    @await RenderSectionAsync("HeadMeta", required: false)
</head>
<body dir="@Orchard.CultureDir()">
    <nav class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
        <div class="container">
            <shape type="Branding" />
            <button type="button" class="navbar-toggler" data-bs-toggle="collapse" data-bs-target="#navbar" aria-expanded="false" aria-controls="navbar" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="navbar">
                <menu alias="alias:main-menu" cache-id="main-menu" cache-fixed-duration="00:05:00" cache-tag="alias:main-menu" />
                <shape type="LoginMenu" />
            </div>
        </div>
    </nav>
    @await RenderSectionAsync("Header", required: false)
    <main class="container">
        @await RenderSectionAsync("Messages", required: false)
        @await RenderBodyAsync()
    </main>
    @if (IsSectionDefined("Footer"))
    {
        <footer>
            <div class="container">
                @await RenderSectionAsync("Footer", required: false)
            </div>
        </footer>
    }

    <!-- This -->
    <shape Type="MyShape" />

    <resources type="FootScript" />
</body>
</html>
```

### Expected behavior
The <style at="Head"> code block defined in Shape should be rendered into the Head of the Html document

### Screenshots

![image](https://github.com/OrchardCMS/OrchardCore/assets/32150676/7624c5ab-3b3d-4225-8d87-8e8d09b1abaf)

![image](https://github.com/OrchardCMS/OrchardCore/assets/32150676/fe5a45f0-3d8c-49f7-b10e-224662189aaf)

![image](https://github.com/OrchardCMS/OrchardCore/assets/32150676/089ba629-a841-4df2-be61-90f2d17bbeeb)

![image](https://github.com/OrchardCMS/OrchardCore/assets/32150676/7422c073-9b64-4470-a5dd-beb0d52d20c5)

**Comment by sebastienros** [comment]
Is it possible that you haven't included the tag helpers for `<style>` in your view imports ?

**Comment by MikeAlhayek** [comment]
@devenliu is this still an issue?

**Comment by devenliu** [comment]
> Is it possible that you haven't included the tag helpers for `<style>` in your view imports ?

Thank you very much for your reply. 
The relevant namespace has been imported in `_ViewImports.cshtml`. 
At the same time, you can also know from the screenshot that the <style> tag is purple (this is the capability of VisualStudio), which means that it It is effective.

**Comment by jtkech** [comment]
When the `Layout.cshtml` is executed it's too late, all theme zones (we also have content item zones) have already been processed.

 Most of shapes are driven by content/part/field drivers and are executed when the Theme "Content" zone (the body) is processed. All these shapes can be overridden e.g. in a Theme by using Alternates more or less specific for example by using content type/part/field names.

So your `<shape Type="MyShape" />` needs to be part of one of the above alternate shapes, or be driven by a custom driver. And if you don't want to render it in the Theme "Content" zone you can use the `<zone name="SomeZone">` tag helper, see below in the example.

We also have Widgets that we can add in different Theme zones defined in the Admin but we don't provide a Razor widget, you would need a Liquid widget (having a LiquidPart) and use Liquid syntax for which we have zone and shape liquid filters.

For example if we override the `Content.cshtml` file and add the following at the end.

    @using OrchardCore.Mvc.Utilities

    @{

        Model.Classes.Add("content-item");
        ...
    }

    <article class="@String.Join(" ", Model.Classes.ToArray())">
        @if (Model.Header != null || Model.Meta != null) // ContentItem zone.
        {
            ...
        }
        @await DisplayAsync(Model.Content) // ContentItem Content zone.
        @if (Model.Footer != null) // ContentItem zone.
        {
            ...
        }
    </article>

    // To be rendered in the Theme Footer zone,
    // Footer needs to be added to the Zones in the Admin.
    <zone name="Footer">
        <shape Type="MyShape" /> // Your shape.
    </zone>

We get what you were expecting.

![MyShape](https://github.com/OrchardCMS/OrchardCore/assets/8586360/7f106e21-0ec0-4460-86dd-de59c92f1854)

**Comment by jtkech** [comment]
Yes, much more simple!

**Comment by devenliu** [comment]
@ns8482e 
So, if I want to render a custom Shape in a Layout, I have to wrap the Shape inside a Zone, or inside a Content/Widget Template, right?

**Comment by ns8482e** [comment]
@devenliu No! Not completely true ! You can render shape in layout without wrapping zone in layout.

However here your issue is: your shape rendering in layout happens after stylesheet resources rendering and your shape is now registering a resource that is nowhere to be rendered. Wrapping your shape within zone and putting it before stylesheet will register your css resource before it's rendered and your shape will rendered at place where your zone is rendered.

Ideally you should place your shape to zone from display driver as @jtkech suggests and you define zones in layout and render zones in Layout.

**Comment by devenliu** [comment]
@ns8482e 
You are right, I understand it somewhat, but compared with the built-in Layout.cshtml mechanism of ASP.NET Core, I think this is not intuitive enough and not easy to understand. After all, I have declared <resources type= in advance in Layout. "Stylesheet" />.
So,
